File: app/src/main/java/com/antonio/samir/meteoritelandingsspots/service/server/MeteoriteServerResult.java
package com.antonio.samir.meteoritelandingsspots.service.server;

import com.antonio.samir.meteoritelandingsspots.model.Meteorite;

import java.util.List;

public class MeteoriteServerResult {
    private List<Meteorite> meteorites;
    private MeteoriteServerException exception;

    public void setMeteorites(List<Meteorite> meteorites) {
        this.meteorites = meteorites;
    }

    public void setException(MeteoriteServerException exception) {
        this.exception = exception;
    }

    public List<Meteorite> getMeteorites() throws MeteoriteServerException {
        if (exception != null) {
            throw exception;
        }
        return meteorites;
    }

    public MeteoriteServerException getException() {
        return exception;
    }
}

File: app/src/main/java/com/antonio/samir/meteoritelandingsspots/service/local/MeteoriteServerResult.kt
package com.antonio.samir.meteoritelandingsspots.service.local

import com.antonio.samir.meteoritelandingsspots.model.Meteorite

class MeteoriteServerResult {
    var meteorites: List<Meteorite>? = null
        @Throws(MeteoriteServerException::class)
        get() {
            if (exception != null) {
                throw exception as MeteoriteServerException
            }
            return field
        }
    private var exception: MeteoriteServerException? = null

    fun setException(exception: MeteoriteServerException) {
        this.exception = exception
    }

}
